### PR TITLE
Prepare a pipeline for publishing releases to pypi

### LINKED ISSUE #58: Make pip installable (publish to pypi)
Needs a handful of changes:

- [x] reorg top level directory so that `recommender` and `simulator` modules are namespaced under `vasim`
- [x] fix `import` rules
- [x] convert to pyproject.toml (#39)
- [ ] add CI pipeline workflow to publish on version/release change

Some other semi-related tasks we've identified:
- [x] ~consolidate to CamelCase instead of a mix of snake_case and CamelCase named files~
      We simply documented the difference in #63.
- [ ] reorg tests to somewhat mirror the src directory

### PR DESCRIPTION
**Pull request by bpkroth**
# Pull Request

## Title

Prepares a Github Actions workflow for publishing releases to pypi.org

---

## Description

- Bumps the patch version to allow a release to happen.
- [x] Document release process.
- Closes #58 
---

## Type of Change

- ✨ New feature  

---

## Testing

- [x] Add tests of the build and install process in the regular CI pipeline.

---

### CODE REVIEW

**Comment by bpkroth** [comment]
This is more or less ready to go, but the secret hasn't been added to Github yet since it's still waiting on pypi: https://github.com/pypi/support/issues/4783

**Comment by ksaur** [comment]
closes #10

**Review by ksaur — APPROVED** [review]
Thanks! Can you do the release on github (add the tag) after it merges? and then we can figure out pypi pushing later 🎊

File: pyproject.toml [hunk 1]
```diff
@@ -28,7 +28,7 @@ maintainers = [
     { name = "VASim Maintainers", email = "vasim-maintainers@service.microsoft.com" },
 ]
 
-version = "0.1.0"
+version = "0.1.1"
```
Comment (bpkroth) [inline pyproject.toml:31]:
```suggestion
version = "0.1.0"
```

leaving version as is for now

**Pull request merged**